Marlin/ultralcd.cpp

```diff
@@ -805,13 +807,29 @@ static void lcd_prepare_menu() {
 
 #endif // DELTA_CALIBRATION_MENU
 
-inline void line_to_current(AxisEnum axis) {
-  #if ENABLED(DELTA)
-    calculate_delta(current_position);
-    plan_buffer_line(delta[X_AXIS], delta[Y_AXIS], delta[Z_AXIS], current_position[E_AXIS], manual_feedrate[axis]/60, active_extruder);
-  #else
-    plan_buffer_line(current_position[X_AXIS], current_position[Y_AXIS], current_position[Z_AXIS], current_position[E_AXIS], manual_feedrate[axis]/60, active_extruder);
-  #endif
+/**
+ * If the manual move hasn't been fed to the planner yet,
+ * and the planner can accept one, send immediately
+ */
+inline void manage_manual_move() {
+  if (manual_move_axis != (int8_t)NO_AXIS && !planner_is_full()) {
+    #if ENABLED(DELTA)
```

Comment:
The stepper ISR code needs to be light and fast, but portions of the code that are written to run in the main `loop()` should just aim for simplicity and saving bytes where possible.

Still, do you think it would help to `FORCE_INLINE` the `planner_is_full` function and move its definition to planner.cpp? Apart from not being inline it shouldn't have that mich overhead compared to `movesplanned()`. They both return a `char`.
```cpp
FORCE_INLINE uint8_t movesplanned() { return BLOCK_MOD(block_buffer_head - block_buffer_tail + BLOCK_BUFFER_SIZE); }

bool planner_is_full() { return (block_buffer_tail == next_block_index(block_buffer_head)); }
bool planner_is_full() { return (block_buffer_tail == BLOCK_MOD(block_buffer_head + 1)); }

FORCE_INLINE int8_t next_block_index(int8_t block_index) { return BLOCK_MOD(block_index + 1); }
```